cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -98,6 +105,39 @@ func (w *Watcher) check() {
 func (w *Watcher) RecordUsage(e entry) {
 	logging.Debug("Recording usage of %s (%d)", e.Exec, e.PID)
 	w.an.Event(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, e.Dims)
+	w.UpdateLastUsed(e)
+}
+
+func (w *Watcher) UpdateLastUsed(e entry) {
+	logging.Debug("Updating last usage of project that contains %s", e.Exec)
+	localProjects := projectfile.GetProjectMapping(w.cfg)
+	for namespace, checkouts := range localProjects {
+		for _, checkout := range checkouts {
+			proj, err := project.FromPath(checkout)
+			if err != nil {
+				multilog.Error("Unable to get project %s from checkout: %v", checkout, err)
+				continue
+			}
+			projectTarget := target.NewProjectTarget(proj, nil, "")
+			execDir := setup.ExecDir(projectTarget.Dir())
+			logging.Debug("Looking at project %s located at %s whose executables are in %s", namespace, checkout, execDir)
+			if !strings.HasPrefix(e.Exec, execDir) {
```

Comment:
We'll not to do something more sophisticated here to handle different representations of the same path. See https://github.com/ActiveState/cli/blob/79cb7134123e7836b1c9fd52111dd274aea956e4/internal/fileutils/fileutils.go#L1157-L1157 for example.